src/components/entity/upload/table.vue

```diff
@@ -0,0 +1,166 @@
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <div ref="container" class="entity-upload-table"
+    :class="{ 'overlaps-popup': overlapsPopup }" :style="{ minHeight }">
+    <table class="table">
+      <thead>
+        <tr>
+          <th><span class="sr-only">{{ $t('common.rowNumber') }}</span></th>
+          <th>label</th>
+          <th v-for="{ name } of dataset.properties" :key="name">
+            <div v-tooltip.text>{{ name }}</div>
+          </th>
+        </tr>
+      </thead>
+      <tbody v-if="entities != null && entities.length !== 0"
+        :class="{ 'data-loading': awaitingResponse }">
+        <tr v-for="(entity, entityIndex) in entities" :key="entityIndex">
+          <td class="row-number">
+            {{ $n(rowIndex + entityIndex + 1, 'noGrouping') }}
+          </td>
+          <td><div v-tooltip.text>{{ entity.label }}</div></td>
+          <td v-for="{ name } of dataset.properties" :key="name">
+            <div v-tooltip.text>{{ entity.data[name] }}</div>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script setup>
+import { nextTick, ref, watch } from 'vue';
+
+import { px } from '../../../util/dom';
+import { useRequestData } from '../../../request-data';
+
+defineOptions({
+  name: 'EntityUploadTable'
+});
+const props = defineProps({
+  entities: Array,
+  // The 0-indexed row number of the first row of the table
+  rowIndex: Number,
+  pageSize: {
+    type: Number,
+    required: true
+  },
+  awaitingResponse: Boolean
+});
+
+// The component assumes that this data will exist when the component is
+// created.
+const { dataset } = useRequestData();
+
+// The table container
+const container = ref(null);
+
+// Prevent the table container from shrinking on the last page.
+const minHeight = ref('auto');
+// The height of the last full page of the current size. Equals 0 when there is
+// no height to use.
+let previousHeight = 0;
+watch(
+  [() => props.entities, () => props.pageSize],
+  ([entities, newSize], [, oldSize]) => {
+    // Reset minHeight.
+    minHeight.value = 'auto';
+
+    // If the page size has changed, reset previousHeight, as it is no longer
+    // useful. If entities == null without the page size changing, that means
+    // that there has been a change like the modal being hidden that should
+    // cause previousHeight to be reset.
+    if (newSize !== oldSize || entities == null) previousHeight = 0;
+
+    // Either set or attempt to use previousHeight.
+    if (entities != null && entities.length === newSize) {
+      nextTick(() => {
+        previousHeight = container.value.getBoundingClientRect().height;
+      });
+    } else if (previousHeight !== 0) {
+      minHeight.value = px(previousHeight);
+    }
+  }
+);
+
+const overlapsPopup = ref(false);
+const resizeLastColumn = () => {
```

Comment:
I'm seeing an issue that may have to do with this function:

1. Set up the modal such that when a file is selected, the pop-up will overlap the second table. For example, select a larger page size for the first table.
2. (Not sure whether this matters:) Use an entity with fewer properties, such that the tables don't scroll horizontally, and the last column ends up being wider than the other columns.
3. Select a file. Notice that a scrollbar appears below the second table. Once the animation ends, the scrollbar disappears again. A scrollbar should never appear.
    - Maybe another variable is the width of the viewport?

Are we definitely waiting for the animation to end before calling this function?